plugins/backend/catalog-backend-module-okta/src/providers/OktaGroupProvider.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2021 Larder Software Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { GroupEntity } from '@backstage/catalog-model';
+import * as winston from 'winston';
+import { Config } from '@backstage/config';
+import { OktaEntityProvider } from './OktaEntityProvider';
+
+/**
+ * Provides entities from Okta Group service.
+ */
+export class OktaGroupProvider extends OktaEntityProvider {
+  static fromConfig(config: Config, options: { logger: winston.Logger }) {
+    const orgUrl = config.getString('orgUrl');
+    const token = config.getString('token');
+
+    return new OktaGroupProvider({ orgUrl, token }, options);
+  }
+
+  getProviderName(): string {
+    return `okta-group-${this.orgUrl}`;
+  }
+
+  async run(): Promise<void> {
+    if (!this.connection) {
+      throw new Error('Not initialized');
+    }
+
+    this.logger.info(
+      `Providing okta group resources from okta: ${this.orgUrl}`,
+    );
+    const groupResources: GroupEntity[] = [];
+
+    const client = this.getClient();
+
+    const defaultAnnotations = await this.buildDefaultAnnotations();
+
+    await client.listGroups().each(async group => {
+      const members: string[] = [];
+      await group.listUsers().each(user => {
+        members.push(user.id);
```

Comment:
I'm wondering what the use case is for using the group provider and not the user provider? In the case that we don't use the user provider does it make sense to add members by id here? (these users probably don't exist with that id in the catalog in that case). 

This is very much debatable I'm mostly just curious

Reply:
I guess we could add both, no harm. But it also is a bit tidy having the separate. But I could persuaded either way.

Reply:
If we leave it as two providers its hard to know if the group provider is enabled and the user one is now. If the user one is disabled and the group one is enabled it does show the users as members when you look at the yaml, but it doesnt show in the ui.

Reply:
Myself and Brian had a chat to say that we can add a facade which wraps both in the future if desired.